src/waitlist.cpp

```diff
@@ -95,36 +134,25 @@ bool WaitingList::clientLogin(const Player* player)
 		return false;
 	}
 
-	slot = priorityWaitList.size();
+	slot = info->priorityWaitList.size();
 	if (player->isPremium()) {
-		priorityWaitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
+		info->priorityWaitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
 	} else {
-		slot += waitList.size();
-		waitList.emplace_back(OTSYS_TIME() + (getTimeout(slot + 1) * 1000), player->getGUID());
+		slot += info->waitList.size();
```

Comment:
Yeah, of course, a priority queue seems to be the optimal container if the queue wasn't jsut so specific. Indeed using a PQ could allow for more complex algorithms for queueing :)